lisa/trace.py

```diff
@@ -305,16 +280,52 @@ def has_events(self, events):
         else:
             return set(events).issubset(set(self.available_events))
 
+    def set_window(self, t_min=None, t_max=None):
```

Comment:
If `set_window()` returned a new object that is just a view over the trace, the memoized business is not needed anymore, as the view object would rightly carry that information. That also allows good compositions, so functions can further trim down the trace without having to care about who is going to use the object next, and if it's ok to change the window size.

Reply:
I like that approach, but I need to wrap my head around all the implications that has on the uses of the trace view instances.

Reply:
Mmm it seems to be working okay. This is what I had to add to the new `TraceView` class:
```python
        # <window computation here>
        # Import here to avoid a circular dependency issue at import time
        # with lisa.analysis.base
        from lisa.analysis.proxy import AnalysisProxy
        self.analysis = AnalysisProxy(self)

    def __getattr__(self, name):
        return getattr(self.trace, name)

    def df_events(self, event):
        """
        Get a dataframe containing all occurrences of the specified trace event
        in the parsed trace.

        :param event: Trace event name
        :type event: str
        """
        df = self.trace.df_events(event)
        if not df.empty:
            df = df[self.start:self.end]

        return df
```

Since we re-create an `AnalysisProxy`, that takes care of `memoized` for us.